arch/risc-v/src/esp32c3/rom/esp32c3_libc_stubs.h

```diff
@@ -0,0 +1,288 @@
+/*****************************************************************************
+ * arch/risc-v/src/esp32c3/rom/esp32c3_libc_stubs.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ *****************************************************************************/
+
+#ifndef __ARCH_RISCV_SRC_ESP32C3_ROM_ESP32C3_LIBC_STUBS_H
+#define __ARCH_RISCV_SRC_ESP32C3_ROM_ESP32C3_LIBC_STUBS_H
+
+/*****************************************************************************
+ * Included Files
+ *****************************************************************************/
+
+#include <stdio.h>
+
+#include <sys/time.h>
+#include <sys/times.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <signal.h>
+
+#include <nuttx/mutex.h>
+
+#define _lock_t int
+
+_lock_t g_common_mutex;
+_lock_t g_common_recursive_mutex;
+
+mutex_t g_nxlock_common;
+mutex_t g_nxlock_recursive;
+
+/* Forward declaration */
+
+struct _reent;
+
+int _close_r(struct _reent *r, int fd)
+{
+  return close(fd);
+}
+
+int _fstat_r(struct _reent *r, int fd, struct stat *statbuf)
+{
+  return fstat(fd, statbuf);
+}
+
+int _getpid_r(struct _reent *r)
+{
+  return getpid();
+}
+
+int _kill_r(struct _reent *r, int pid, int sig)
```

Comment:
why not move the implementation to source file?

Reply:
The initial ideas was only to have the struct definition in the header file, but I faced some issues when pointing the functions pointer directly to NuttX's libc functions, then decided to create this "wrapper". Since it is just a wrapper I don't see motivation to move it to a new file.

Reply:
it could simply change to esp32c3_libc_stub.c and move related stuff to this file.

Reply:
Ok, I'll do it